### PR TITLE
Increase swapper id limit, allowing more files (like indexes) to be opened by the page cache, and fix a MemoryManager bug around alignment handling.

### PR DESCRIPTION
**Pull request by chrisvest**
This is back-porting fixes from the #11668 PR, and should be null-merged into 3.4.

### CODE REVIEW

File: community/io/src/main/java/org/neo4j/io/pagecache/impl/muninn/PageList.java [hunk 1]
```diff
@@ -323,12 +313,7 @@ public void initBuffer( long pageRef )
 
     private byte getUsageCounter( long pageRef )
     {
-        return UnsafeUtil.getByteVolatile( offUsage( pageRef ) );
-    }
-
-    private void setUsageCounter( long pageRef, byte count )
-    {
-        UnsafeUtil.putByteVolatile( offUsage( pageRef ), count );
+        return (byte) (UnsafeUtil.getLongVolatile( offFilePageId( pageRef ) ) & 0x07);
```
Comment (MishaDemianenko) [inline community/io/src/main/java/org/neo4j/io/pagecache/impl/muninn/PageList.java]: would be really nice to get constant for mister 0x07

File: community/io/src/main/java/org/neo4j/io/pagecache/impl/muninn/PageList.java [hunk 2]
```diff
@@ -337,11 +322,18 @@ private void setUsageCounter( long pageRef, byte count )
...
         if ( usage < 4 ) // avoid cache sloshing by not doing a write if counter is already maxed out
         {
             usage++;
-            setUsageCounter( pageRef, usage );
+            // Use compareAndSwapLong to only actually store the updated count if nothing else changed
+            // in this word-line. The word-line is shared with the file page id, and the swapper id.
+            // Those fields are updated under guard of the exclusive lock, but we *might* race with
+            // that here, and in that case we would never want a usage counter update to clobber a page
+            // binding update.
+            UnsafeUtil.compareAndSwapLong( null, address, v, (v & 0xFFFFFFFF_FFFFFFF8L) + usage );
```
Comment (MishaDemianenko) [inline community/io/src/main/java/org/neo4j/io/pagecache/impl/muninn/PageList.java]: and for mister xFFFFFFFF_FFFFFFF8L

File: community/io/src/main/java/org/neo4j/io/pagecache/impl/muninn/PageList.java [hunk 3]
```diff
@@ -351,20 +343,21 @@ public void incrementUsage( long pageRef )
...
         }
         return usage == 0;
     }
 
     public long getFilePageId( long pageRef )
     {
-        int highByte = UnsafeUtil.getByte( offFilePageId( pageRef ) ) & UNSIGNED_BYTE_MASK;
-        long lowInt = UnsafeUtil.getInt( offFilePageId( pageRef ) + Byte.BYTES ) & UNSIGNED_INT_MASK;
-        long filePageId = (((long) highByte) << Integer.SIZE) | lowInt;
+        long filePageId = UnsafeUtil.getLong( offFilePageId( pageRef ) ) >>> 24;
```
Comment (MishaDemianenko) [inline community/io/src/main/java/org/neo4j/io/pagecache/impl/muninn/PageList.java]: and for mister 24

**Review by MishaDemianenko — CHANGES_REQUESTED** [review]
couple of minor comments

**Comment by chrisvest** [comment]
@MishaDemianenko 👍

**Review by MishaDemianenko — APPROVED** [review]

**Pull request merged**